chia/rpc/farmer_rpc_api.py

```diff
@@ -218,23 +218,23 @@ async def _state_changed(self, change: str, change_data: Optional[Dict[str, Any]
         return payloads
 
     async def get_signage_point(self, request: Dict[str, Any]) -> EndpointResult:
-        sp_hash = hexstr_to_bytes(request["sp_hash"])
-        for _, sps in self.service.sps.items():
-            for sp in sps:
-                if sp.challenge_chain_sp == sp_hash:
```

Comment:
this code seems to suggest that there may be signage points recorded under the wrong hash. Is that just someone not understanding how the dictionary works, or is this a real concern?
Did you dig into the history of where this comes from?

Reply:
you make the assumption that this linear scan of all signage points, looking for `sp_hash` behaves the same as your version, where you only pick the *first* signage point. Is this list guaranteed to only contain 1 item?
Did you make sure that we always insert SPs under the right key in this dict?